engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala

```diff
@@ -62,21 +67,41 @@ class LanguageServerComponent(config: LanguageServerConfig, logLevel: LogLevel)
     } yield ComponentStarted
   }
 
+  /** Start the application sampling. */
+  private def startSampling(config: LanguageServerConfig): MethodsSampler = {
+    val sampler = config.profilingConfig.profilingPath match {
+      case Some(path) =>
+        val s = new FileSampler(path.toFile)
+        JavaLoggingLogHandler.registerLogFile(s.getSiblingFile(".log"))
+        s
+      case None =>
+        NoopSampler()
+    }
+    sampler.start()
+    config.profilingConfig.profilingTime.foreach(sampler.stop(_))
+
+    sampler
+  }
+
   /** @inheritdoc */
   override def stop(): Future[ComponentStopped.type] =
     maybeServerCtx match {
       case None =>
-        Future.failed(new Exception("Server isn't running"))
+        Future.successful(ComponentStopped)
 
       case Some(serverContext) =>
         for {
+          _ <- stopSampling(serverContext)
```

Comment:
Yup, this looks like fine order of actions. Stopping sampling before terminating Truffle.